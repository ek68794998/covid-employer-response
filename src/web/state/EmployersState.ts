import { EmployerRecord } from "../../common/EmployerRecord";

export interface EmployersState {
	items: EmployerRecord[];
}