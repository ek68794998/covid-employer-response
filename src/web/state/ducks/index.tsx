import { combineReducers } from "redux";

import { getLocalizedStrings } from "./localization/reducers";

export default combineReducers({
	strings: getLocalizedStrings,
});