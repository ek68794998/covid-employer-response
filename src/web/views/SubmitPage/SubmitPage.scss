@import "../Colors.scss";
@import "../Dimensions.scss";

$body-vert-width: 990px;
$card-padding: 16px;
$github-color: #24292e;
$google-forms-color: #673ab7;
$icon-size: 32px;

main#submit {
	margin: 0 auto;
	width: 100%;

	.SubmitPage__Container {
		padding: 0 $page-padding;
	}

	.SubmitPage__Content {
		line-height: 180%;
		margin: 0 auto $page-padding;
		max-width: $body-max-width;
	}

	.SubmitPage__Options {
		display: flex;

		.SubmitPage__Option {
			background: $background;
			border: 1px solid $outline-color;
			border-radius: $card-padding;
			color: transparentize($color: $header, $amount: 0.3);
			display: flex;
			flex: 1;
			flex-direction: column;
			padding: $card-padding;
			transition: 0.1s all;

			&:hover {
				box-shadow: 0px 3px 6px $card-shadow-color;
				color: $background;
				transform: translateY(-1px);

				&.SubmitPage__Option--DataForm {
					background: $google-forms-color;
					border-color: $google-forms-color;
				}

				&.SubmitPage__Option--DataGitHub, &.SubmitPage__Option--CodeGitHub {
					background: $github-color;
					border-color: $github-color;
				}
			}

			h2 {
				margin: 0;
			}

			.SubmitPage__OptionIcon {
				background: rgba(0, 0, 0, 0.4);
				border-radius: ($icon-size / 2 + $card-padding);
				line-height: 0;
				margin: 0 auto;
				padding: ($card-padding / 2);

				img {
					height: $icon-size;
					width: $icon-size;
				}
			}
		}

		.SubmitPage__Spacer {
			height: 24px;
			width: 24px;
		}
	}
}

@media (max-width: $body-vert-width) {
	main#submit {
		.SubmitPage__Content {
			margin-bottom: $page-padding-narrow;
		}

		.SubmitPage__Options {
			flex-direction: column;
		}
	}
}

@media (max-width: $body-narrow-width) {
	main#submit {
		.SubmitPage__Container {
			padding: 0 $page-padding-narrow;
		}
	}
}

@media (any-hover: none) {
	main#submit {
		.SubmitPage__Options {
			.SubmitPage__Option {
				color: $background;

				&.SubmitPage__Option--DataForm {
					background: $google-forms-color;
					border-color: $google-forms-color;
				}

				&.SubmitPage__Option--DataGitHub, &.SubmitPage__Option--CodeGitHub {
					background: $github-color;
					border-color: $github-color;
				}
			}
		}
	}
}
