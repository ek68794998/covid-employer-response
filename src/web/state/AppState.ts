import { EmployersState } from "./EmployersState";

import { LocalizedStrings } from "../../common/LocalizedStrings";

export interface AppState {
	employers: EmployersState;

	environment: string;

	strings: LocalizedStrings;
}