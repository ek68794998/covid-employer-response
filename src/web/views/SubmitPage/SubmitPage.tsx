import React from "react";
import ReactMarkdown from "react-markdown";
import { useSelector } from "react-redux";
import { withRouter } from "react-router-dom";

import { format, LocalizedStrings } from "../../../common/LocalizedStrings";
import { ProjectIssueSubmissionUrl, ProjectUrl, SubmissionFormUrl } from "../../../common/constants/UrlConstants";

import { getStrings } from "../../state/ducks/localization/selectors";

import AppHelmet from "../AppHelmet/AppHelmet";

import "./SubmitPage.scss";

const SubmitPage: React.FC = (): React.ReactElement => {
	const strings: LocalizedStrings = useSelector(getStrings);

	return (
		<main id="submit">
			<AppHelmet
				title={strings.submit}
			/>
			<div className="SubmitPage__Container">
				<div className="SubmitPage__Content">
					<h1>{strings.submitTitle}</h1>
					<ReactMarkdown source={format(strings.submitSubtitle, { appTitle: strings.appTitle })} />
					<section className="SubmitPage__Options">
						<a
							className="SubmitPage__Option SubmitPage__Option--DataForm"
							href={SubmissionFormUrl}
							rel="noopener noreferrer"
							target="_blank"
						>
							<h2>{strings.submitDataForm.title}</h2>
							<ReactMarkdown source={strings.submitDataForm.description} />
							<div className="SubmitPage__OptionIcon"><img src="/images/google-forms.svg" /></div>
						</a>
						<div className="SubmitPage__Spacer" />
						<a
							className="SubmitPage__Option SubmitPage__Option--DataGitHub"
							href={ProjectIssueSubmissionUrl}
							rel="noopener noreferrer"
							target="_blank"
						>
							<h2>{strings.submitDataGithub.title}</h2>
							<ReactMarkdown source={strings.submitDataGithub.description} />
							<div className="SubmitPage__OptionIcon"><img src="/images/github.svg" /></div>
						</a>
						<div className="SubmitPage__Spacer" />
						<a
							className="SubmitPage__Option SubmitPage__Option--CodeGitHub"
							href={ProjectUrl}
							rel="noopener noreferrer"
							target="_blank"
						>
							<h2>{strings.submitCodeGithub.title}</h2>
							<ReactMarkdown source={strings.submitCodeGithub.description} />
							<div className="SubmitPage__OptionIcon"><img src="/images/github.svg" /></div>
						</a>
					</section>
				</div>
			</div>
		</main>
	);
};

export default withRouter(SubmitPage);
