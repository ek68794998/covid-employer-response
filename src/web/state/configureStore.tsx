import { AnyAction, applyMiddleware, compose, createStore, Store } from "redux";
import thunk from "redux-thunk";

import { Production } from "../../common/constants/EnvironmentConstants";

import { AppState } from "./AppState";
import rootReducer from "./ducks";

const configureStore = (preloadedState: Partial<AppState>): Store<AppState, AnyAction> => {
	const isProd: boolean = process.env.NODE_ENV === Production;
	const composeEnhancers: any =
		(!isProd && typeof window !== "undefined" && (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

	const store: Store<AppState, AnyAction> = createStore(
		rootReducer,
		preloadedState,
		composeEnhancers(applyMiddleware(thunk)),
	);

	if (module.hot) {
		// Enable Webpack hot module replacement for reducers
		module.hot.accept("./ducks", () => {
			const nextRootReducer: any = require("./ducks").default;
			store.replaceReducer(nextRootReducer);
		});
	}

	return store;
};

export default configureStore;