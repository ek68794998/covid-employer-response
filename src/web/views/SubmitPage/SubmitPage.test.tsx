import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import { AnyAction, Store } from "redux";

import { getPlocStringsAsync, mockComponent } from "../../../__tests__/TestUtils";

import { AppState } from "../../state/AppState";
import configureStore from "../../state/configureStore";

import SubmitPage from "./SubmitPage";

jest.mock(
	"react-markdown",
	() => mockComponent("ReactMarkdown"));

describe("<SubmitPage />", () => {
	test("renders without exploding", async () => {
		const store: Store<AppState, AnyAction> = configureStore({ strings: await getPlocStringsAsync() });

		const renderedValue: ReactTestRendererJSON | null =
			renderer.create(
				<Provider store={store}>
					<BrowserRouter>
						<SubmitPage />
					</BrowserRouter>
				</Provider>,
			).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
