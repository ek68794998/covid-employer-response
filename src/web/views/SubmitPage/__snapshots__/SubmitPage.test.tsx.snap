// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitPage /> renders without exploding 1`] = `
<main
  id="submit"
>
  <div
    className="SubmitPage__Container"
  >
    <div
      className="SubmitPage__Content"
    >
      <h1>
        §úβ₥ïƭƭïñϱ ñèω ïñƒôř₥áƭïôñ
      </h1>
      [component: ReactMarkdown source=(**ÇôřôñáƲïřúƨ É₥ƥℓô¥èř Rèƨƥôñƨè Třáçƙèř** ïƨ çô₥ƥℓèƭèℓ¥ ôƥèñ-δáƭá áñδ ôƥèñ-ƨôúřçè. Wè ωôúℓδ ℓôƲè ƭô áççèƥƭ ¥ôúř ñèω è₥ƥℓô¥èř èñƭřïèƨ, ñèω δèƭáïℓƨ ôř çℓáï₥ƨ, ôř ¥ôúř çôδè çλáñϱèƨ. HáƲè á ℓôôƙ áƭ ƭλè ôƥƭïôñƨ βèℓôω ƭô ƨèè λôω ¥ôú çáñ λèℓƥ ƨλáƥè ôúř ƒúƭúřè!) /]
      <section
        className="SubmitPage__Options"
      >
        <a
          className="SubmitPage__Option SubmitPage__Option--DataForm"
          href="https://forms.gle/EEcp88X6sf1ocfuH8"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2>
            §úβ₥ïƭ δáƭá Ʋïá ƒôř₥
          </h2>
          [component: ReactMarkdown source=(Tλè èáƨïèƨƭ ωá¥ ƭô ƨúβ₥ïƭ ñèω è₥ƥℓô¥èřƨ ôř èδïƭ èжïƨƭïñϱ ôñèƨ ïƨ ƭλřôúϱλ ôúř **ôñℓïñè Gôôϱℓè ƒôř₥**. Júƨƭ ƒïℓℓ ôúƭ á ƒèω ƒïèℓδƨ áñδ ωè'ℓℓ ₥áƙè ƭλè çλáñϱèƨ βèλïñδ ƭλè ƨçèñèƨ.) /]
          <div
            className="SubmitPage__OptionIcon"
          >
            <img
              src="/images/google-forms.svg"
            />
          </div>
        </a>
        <div
          className="SubmitPage__Spacer"
        />
        <a
          className="SubmitPage__Option SubmitPage__Option--DataGitHub"
          href="https://github.com/ekumlin/covid-employer-response/issues/new/choose"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2>
            §úβ₥ïƭ δáƭá Ʋïá GïƭHúβ
          </h2>
          [component: ReactMarkdown source=(Âñôƭλèř ωá¥ ƭô áδδ ôř úƥδáƭè è₥ƥℓô¥èřƨ ïƨ ƭλřôúϱλ ôúř **GïƭHúβ ïƨƨúèƨ ƥáϱè**. Júƨƭ ôƥèñ áñ ïƨƨúè ôř ₥áƙè ƭλè çλáñϱè ¥ôúřƨèℓƒ áñδ ƨƭářƭ á ƥúℓℓ řè9úèƨƭ.) /]
          <div
            className="SubmitPage__OptionIcon"
          >
            <img
              src="/images/github.svg"
            />
          </div>
        </a>
        <div
          className="SubmitPage__Spacer"
        />
        <a
          className="SubmitPage__Option SubmitPage__Option--CodeGitHub"
          href="https://github.com/ekumlin/covid-employer-response"
          rel="noopener noreferrer"
          target="_blank"
        >
          <h2>
            §úβ₥ïƭ çôδè Ʋïá GïƭHúβ
          </h2>
          [component: ReactMarkdown source=(Wè'δ ℓôƲè ƭô ƨèè ¥ôúř ïδèáƨ ôñ λôω ƭô ƨλáƥè ƭλïƨ ωèβƨïƭè. Júƨƭ ƒôřƙ **ôúř GïƭHúβ řèƥôƨïƭôř¥** áñδ ƨƭářƭ á ƥúℓℓ řè9úèƨƭ ƭô ƨúβ₥ïƭ ¥ôúř çôδè çλáñϱèƨ.) /]
          <div
            className="SubmitPage__OptionIcon"
          >
            <img
              src="/images/github.svg"
            />
          </div>
        </a>
      </section>
    </div>
  </div>
</main>
`;
